### PR TITLE
add more robust gitignore

### PR DESCRIPTION
**Pull request by jaredsburrows**
I keep seeing `.DS_Store` from my MacBook. So I added a more robust `.gitignore` that should handle most cases.

@ben-manes

### CODE REVIEW

**Review by ben-manes — APPROVED** [review]
This looks good except that you dropped the existing ones instead of adding. If you bring those back then lgtm

**Comment by jaredsburrows** [comment]
@ben-manes I merged both of the `.gitignore` files and removed the duplicates.

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,96 @@
...
+# Files for the Dalvik VM
+*.dex
+
+# Java class files
+*.class
+
+# Gradle files
+.gradle/
+build/
+buildSrc/
```
Comment (eygraber) [inline .gitignore:34]: Out of curiosity, why ignore `buildSrc`? That's usually important.

Comment (jaredsburrows) [reply .gitignore:34]: https://github.com/ben-manes/gradle-versions-plugin/pull/605.

**Pull request merged**